org.eclipse.xtext/src/org/eclipse/xtext/serializer/impl/Serializer.java

```diff
@@ -210,7 +213,60 @@ public ReplaceRegion serializeReplacement(EObject obj, SaveOptions options) {
 			throw new IllegalStateException("Cannot replace an obj that has no associated node");
 		}
 		String text = serialize(obj, options);
-		return new ReplaceRegion(node.getTotalOffset(), node.getTotalLength(), text);
+		int replaceRegionLength = calculateReplaceRegionLength(node, text);
+		return new ReplaceRegion(node.getTotalOffset(), replaceRegionLength, text);
 	}
 
+	/**
+	 * @since 2.22
+	 */
+	protected int calculateReplaceRegionLength(ICompositeNode node, String text) {
+		int oldTextLength = node.getTotalLength();
+		int newTextLength = text.length();
+
+		if (newTextLength > oldTextLength) {
+			String remainingText = text.substring(oldTextLength);
+			if (isWhitespace(remainingText) && hiddenNodeFollows(node)) {
+				return newTextLength;
+			}
+		}
+		return oldTextLength;
+	}
+
+	/**
+	 * @since 2.22
+	 */
+	protected boolean hiddenNodeFollows(ICompositeNode node) {
+		INode followingNode = getFollowingNode(node);
+		if (followingNode instanceof ILeafNode) {
+			return ((ILeafNode)followingNode).isHidden();
+		}
+		return false;
+	}
+	
+	/**
+	 * Returns the node that follows the <i>node</i>, independently, if they have the same parent.
+	 * @since 2.22
+	 */
+	protected INode getFollowingNode(ICompositeNode node) {
+		if (node != null) {
+			if (node.hasNextSibling()) {
+				INode nextSibling = node.getNextSibling();
+				Iterator<ILeafNode> nextSiblingLeafNodes = nextSibling.getLeafNodes().iterator();
+				if (nextSiblingLeafNodes.hasNext()) {
+					return nextSiblingLeafNodes.next();
+				} else {
+					return getFollowingNode(node.getParent());
+				}	
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * @since 2.22
+	 */
+	private boolean isWhitespace(String text) {
```

Comment:
Last thing that slipped through: This method should either be protected or the since tag is unnecessary.

Reply:
I changed the method visibility to `protected`.